### PR TITLE
I2c small fixes

### PR DESCRIPTION
**Pull request by anselmolsm**
The first commit fixes a KW complaint.
The second patch is to follow the ioctl doc.

### CODE REVIEW

File: src/shared/sol-i2c-linux.c [hunk 1]
```diff
@@ -99,6 +99,7 @@ sol_i2c_open_raw(uint8_t bus, enum sol_i2c_speed speed)
     return i2c;
 
 open_error:
+    close(i2c->dev);
```
Comment (lpereira) [inline src/shared/sol-i2c-linux.c]: If `open()` fails, this will attempt to close an undefined file descriptor.

Comment (anselmolsm) [inline src/shared/sol-i2c-linux.c]: Ops, true story. I'll fix that. tks.

**Comment by anselmolsm** [comment]
v2: Fix issue pointed out by @lpereira .

**Comment by anselmolsm** [comment]
Applied, tks.

**Pull request closed without merging**